include/vcpkg/metrics.h

```diff
@@ -8,6 +8,68 @@
 
 namespace vcpkg
 {
+    template<typename T>
+    struct MetricEntry
+    {
+        T metric;
+        StringLiteral name;
+    };
+
+    enum class DefineMetric
+    {
+        AssetSource,
```

Comment:
Are these referred to more than once? If not, I think the previous form was actually better since we didn't have to decode the enum to a string and back again.

If you keep this, we probably should have a comment about how adding new entries needs to be classified vis a vis GDPR and all that stuff?